lib/matplotlib/__init__.py

```diff
@@ -547,6 +548,11 @@ def _create_tmp_config_dir():
     tempdir = os.path.join(tempdir, 'matplotlib-%s' % getpass.getuser())
     os.environ['MPLCONFIGDIR'] = tempdir
 
+    try:
+        mkdirs(tempdir)
+    except OSError:
```

Comment:
I would not eat the exception here, this leaves the user in the same situation as before if this fails.